src/repo-service.cpp

```diff
@@ -11,29 +14,134 @@
 #include "ui/main-window.h"
 #include "ui/download-repo-dialog.h"
 #include "utils/utils.h"
+#include "rpc/rpc-client.h"
 
 #include "repo-service.h"
 #include "repo-service-helper.h"
 
 namespace {
 
+class SyncedSubfolder {
+public:
+    SyncedSubfolder() {}
+    SyncedSubfolder(const QString &repo_id, const QString &parent_repo_id,
+                    const QString &parent_path)
+        : repo_id_(repo_id), parent_repo_id_(parent_repo_id),
+          parent_path_(parent_path) {}
+    SyncedSubfolder(const ServerRepo &repo, const ServerRepo &parent_repo,
+                    const QString &parent_path)
+        : repo_id_(repo.id), parent_repo_id_(parent_repo.id),
+          parent_path_(parent_path) {}
+    SyncedSubfolder(const SyncedSubfolder &rhs)
+        : repo_id_(rhs.repo_id_), parent_repo_id_(rhs.parent_repo_id_),
+          parent_path_(rhs.parent_path_) {}
+
+    SyncedSubfolder& operator=(const SyncedSubfolder &rhs)
+    {
+        repo_id_ = rhs.repo_id_;
+        parent_repo_id_ = rhs.parent_repo_id_;
+        parent_path_ = rhs.parent_path_;
+        return *this;
+    }
+
+    const QString& repo_id() const { return repo_id_; }
+    const QString& parent_repo_id() const { return parent_repo_id_; }
+    const QString& parent_path() const { return parent_path_; }
```

Comment:
Okay, this class is used for repo-service.cpp internally and can be changed any time.